src/ophyd_async/tango/base_devices/_base_device.py

```diff
@@ -0,0 +1,122 @@
+from __future__ import annotations
+
+from typing import Optional, Union, get_type_hints
+
+from ophyd_async.core import (
+    DEFAULT_TIMEOUT,
+    Device,
+    Signal,
+)
+from ophyd_async.tango.signal import (
+    infer_python_type,
+    infer_signal_frontend,
+    make_backend,
+)
+from tango import DeviceProxy as SyncDeviceProxy
+from tango.asyncio import DeviceProxy as AsyncDeviceProxy
+
+
+class TangoDevice(Device):
```

Comment:
If possible, I think we should keep the general approach to both epics an tango devices in rough step. That does not mean this PR should squeeze a tango-shaped implementation into an epics-shaped hole. 

For example though, there isn't a generic `EpicsDevice` base class, should there be? Is this a result of a difference between the two control systems or developer-preference? Is there some paradigm we can find where the two implementations look roughly similar, and so the codebase is easier to navigate?

I have picked on this particular example but this comment applies to the whole PR.

Reply:
There isn't an `EpicsDevice` because EPICS is decentralized, but there should be a `PviDevice` because there is a single introspection PV which configures the rest of the Device. I think we should concentrate on making `PviDevice` and `TangoDevice` look similar, then we need a `FastCSDevice` that uses the connection mechanism of either of these according to the URI

Reply:
To expand on what Tom said, there is no EpicsDevice because in EPICS, every PV is basically equivalent to every other PV. There is no inherent signal heirarchy. Tango is fundamentally different from EPICS in this respect because instead of individual signals, you interface with a device server which is intended to be a fully functional device containing attributes, functions, etc. Sort of like an IOC if you were to design one for a single device. This means that a device with a Tango signal requires a shared TangoProxy resource for all signals that are connected to the same TangoServer. Currently, Device does not contain an easy way to share a resource across signals so a new TangoDevice is one way to solve this problem.

Reply:
That makes sense to me, the other option is caching of some sort but I leave it to others to decide what is cleaner